collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -53,25 +59,29 @@ public void formLoaded(@NotNull FormController formController) {
 
         boolean hasBackgroundRecording = formController.getFormDef().hasAction(RecordAudioActionHandler.ELEMENT_NAME);
         this.hasBackgroundRecording.setValue(hasBackgroundRecording);
+        updateIndex();
     }
 
     public boolean isFormControllerSet() {
         return formController != null;
     }
 
-    @Nullable
-    public FormIndex getCurrentIndex() {
-        if (formController != null) {
-            return formController.getFormIndex();
-        } else {
-            return null;
-        }
+    public LiveData<FormIndex> getCurrentIndex() {
+        return currentIndex;
```

Comment:
Maintaining this is a bit risky I guess since in multiple places we have to remember about calling `updateIndex()`
Maybe it would be better to add this livedata in `FormController` because there index is update in only one method called `jumpToIndex`. What do you think?

Reply:
I'm torn. I do see your point, but I've been feeling less and less like I want to add to `FormController` as it feels like a pretty clunky wrapper around `FormEntryContoller`. With upcoming entity based work, we might want to sit down and plan out exactly how we should be interacting with JavaRosa from Collect. Are you OK with us leaving this the way it is for now but trying to have a "grander" conversation about it later?